src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexAssemblyCompiler.cs

```diff
@@ -0,0 +1,273 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Threading;
+using System.Reflection;
+using System.Reflection.Emit;
+
+#if DEBUG // until it can be fully implemented
+namespace System.Text.RegularExpressions
+{
+    /// <summary>Compiles a Regex to an assembly that can be saved to disk.</summary>
+    internal sealed class RegexAssemblyCompiler : RegexCompiler
+    {
+        /// <summary>Type count used to augment generated type names to create unique names.</summary>
+        private static int s_typeCount = 0;
+
+        private AssemblyBuilder _assembly;
```

Comment:
It seems like it might read better to pass some of these between the methods instead of having fields. It seems odd to have a BakeMethod that just nulls the field. But just a nit

Reply:
This comes from the original netfx code. I didn't spend much time cleaning it up. I can so some more.